src/mlpack/tests/string_utility_test.cpp

```diff
@@ -0,0 +1,176 @@
+      return retval;
+    }
+    retval = str.substr(0, pos);
+    str.remove_prefix(pos + 1);
+  }
+  return retval;
+}
+
+/**
+ * Test dictionary encoding.
+ */
+BOOST_AUTO_TEST_CASE(DictionaryEncodingTest)
+{
+  std::vector<string>arr(3);
+  arr[0] = "hello how are you";
+  arr[1] = "i am good";
+  arr[2] = "Good how are you";
+  arma::sp_mat output;
+  data::DictionaryEncoding en;
+  en.Encode(arr, output, tokenizer);
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>>& maps = en.Mappings();
+  // Checking that everything is mapped to different numbers
+  std::unordered_map<size_t, size_t>cnt;
+  for (auto it = maps.begin(); it != maps.end(); it++)
+  {
+    cnt[it->second]++;
+    // Every token should be mapped only once
+    BOOST_REQUIRE_EQUAL(cnt[it->second], 1);
+  }
+}
+
+/**
+* Test for SplitByChar class.
+*/
+BOOST_AUTO_TEST_CASE(SplitByCharTest)
+{
+  std::vector<boost::string_view> tokens;
+  boost::string_view strview = "hello how are you" ;
+  SplitByChar obj(" ");
+  boost::string_view token;
+  token = obj(strview);
+  while (!token.empty())
+  {
+    tokens.push_back(token);
+    token = obj(strview);
+  }
+  BOOST_REQUIRE_EQUAL(tokens[0], "hello");
+  BOOST_REQUIRE_EQUAL(tokens[1], "how");
+  BOOST_REQUIRE_EQUAL(tokens[2], "are");
+  BOOST_REQUIRE_EQUAL(tokens[3], "you");
+}
+
+/**
+* Test Dictionary encoding for characters using lamda function.
+*/
+BOOST_AUTO_TEST_CASE(DictionaryEncodingCharTest)
+{
+  std::vector<string>arr(3);
+  arr[0] = "GACCA";
+  arr[1] = "ABCABCD";
+  arr[2] = "GAB";
+  arma::sp_mat output;
+  data::DictionaryEncoding en;
+  // Passing a empty string to encode characters
+  en.Encode(arr, output, [](boost::string_view& str) {
+      if (str.empty())
+        return str;
+      boost::string_view retval = str.substr(0, 1);
+      str.remove_prefix(1);
+      return retval;
+  });
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>>& maps = en.Mappings();
+  // Checking that everything is mapped to different numbers.
+  std::unordered_map<size_t, size_t>cnt;
+  for (auto it = maps.begin(); it != maps.end(); it++)
+  {
+    cnt[it->second]++;
+    // Every token should be mappend only once.
+    BOOST_REQUIRE_EQUAL(cnt[it->second], 1);
+  }
+}
+
+/**
+* Testing the functionality of copy constructor.
+*/
+BOOST_AUTO_TEST_CASE(CopyConstructorCharTest)
+{
+  std::vector<string>arr(3);
+  arr[0] = "hello how are you";
+  arr[1] = "i am good";
+  arr[2] = "Good how are you";
+  arma::sp_mat output;
+  data::DictionaryEncoding en;
+  en.Encode(arr, output, SplitByChar(" "));
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>>& maps = en.Mappings();
+  data::DictionaryEncoding en2 = en;
+  const std::unordered_map<boost::string_view, size_t,
+      boost::hash<boost::string_view>>& maps2 = en2.Mappings();
+  // Comparing both of them.
+  BOOST_REQUIRE_EQUAL(maps2 == maps, true);
```

Comment:
I think `BOOST_REQUIRE_EQUAL(maps2, maps);` is better.

Reply:
I think so , it doesn't support, Throws up big chuck error, which i am unable to debug .